src/services/models/MediaType.ts

```diff
@@ -57,35 +63,93 @@ export class MediaTypeModel {
       skipNonRequired: this.isRequestType && this.onlyRequiredInSamples,
       maxSampleDepth: this.generatedPayloadSamplesMaxDepth,
     };
-    if (this.schema && this.schema.oneOf) {
-      this.examples = {};
-      for (const subSchema of this.schema.oneOf) {
-        const sample = Sampler.sample(subSchema.rawSchema as any, samplerOptions, parser.spec);
+    if (this.schema) {
+      if (this.schema.oneOf) {
+        this.examples = {};
+        for (const subSchema of this.schema.oneOf) {
+          const sample = Sampler.sample(subSchema.rawSchema as any, samplerOptions, parser.spec);
 
-        if (this.schema.discriminatorProp && typeof sample === 'object' && sample) {
-          sample[this.schema.discriminatorProp] = subSchema.title;
+          if (this.schema.discriminatorProp && typeof sample === 'object' && sample) {
+            sample[this.schema.discriminatorProp] = subSchema.title;
+          }
+
+          this.examples[subSchema.title] = new ExampleModel(
+            parser,
+            {
+              value: sample,
+            },
+            this.name,
+            info.encoding,
+          );
+
+          const xmlExamples = this.resolveXmlExample(parser, sample as OpenAPIExample);
+          if (xmlExamples[0]) {
+            this.examples[subSchema.title].value = xmlExamples[0]?.exampleValue;
```

Comment:
`xmlExamples[0]?` this check seems to be redundant as the same is checked in `if`